valhalla/sif/dynamiccost.h

```diff
@@ -868,13 +868,38 @@ class DynamicCost {
 };
 
 using cost_ptr_t = std::shared_ptr<DynamicCost>;
+using mode_costing_t = std::array<cost_ptr_t, static_cast<size_t>(TravelMode::kMaxTravelMode)>;
 
 /**
  * Parses the cost options from json and stores values in pbf.
  * @param object The json request represented as a DOM tree.
  * @param pbf_costing_options A mutable protocol buffer where the parsed json values will be stored.
  */
-void ParseCostOptions(const rapidjson::Value& obj, CostingOptions* pbf_costing_options);
+void ParseSharedCostOptions(const rapidjson::Value& obj, CostingOptions* pbf_costing_options);
+
+/**
+ * Parses all the costing options for all supported costings
+ * @param doc                   json document
+ * @param costing_options_key   the key in the json document where the options are located
+ * @param options               where to store the parsed costing options
+ */
+void ParseCostingOptions(const rapidjson::Document& doc,
+                         const std::string& costing_options_key,
+                         Options& options);
+
+/**
+ * Parses the costing options for the costing specified within the json object. If the
+ * json object has no key named "costing" the type of costing cannot be found and an
+ * exception is thrown
+ * @param doc                   json document
+ * @param key                   the key in the json document where the options are located
+ * @param costing_options       where to store the parsed options
+ * @param costing               specify the costing you want to parse or let it check the json
+ */
+void ParseCostingOptions(const rapidjson::Document& doc,
```

Comment:
this two functions handle the parsing of costing options for the different costings. all this logic used to live inside  src/worker.cc but since its all specific to costing i figured it made sense to move it into this namespace and call into it especially when we needed to do it again for the recosting cost options